src/brace_cleanup.cpp

```diff
@@ -691,7 +691,7 @@ static void parse_cleanup(parse_frame_t *frm, chunk_t *pc)
       LOG_FMT(LSTMT, "%s(%d): %zu> reset expr on %s\n",
               __func__, __LINE__, pc->orig_line, pc->text());
    }
-   else if (pc->type == CT_BRACE_CLOSE)
+   else if (pc->type == CT_BRACE_CLOSE && pc->pp_level == 0)
```

Comment:
There is a test in `scripts` folder that checks if an unmatched brace error is given. You can extend that. If that would have contained the test above this patch would introduce a regression which is not ok.

Your argument about that it's ok that the code will not compile doesn't really hold because you're fixing another case while breaking another.